Fix Linux build warnings

The Linux jobs of the GitHub Actions build show some warnings.

----

GCC warnings:

> ```
> OP2-Landlord/Gui.cpp: In member function 'Gui::AppState Gui::initialSetup()':
> OP2-Landlord/Gui.cpp:76:14: warning: unused variable 'io' [-Wunused-variable]
>    76 |     ImGuiIO& io = ImGui::GetIO();
>       |              ^~
> ```

> ```
> OP2-Landlord/main.cpp: In function 'void mainLoop(EditorConfig&, Graphics&, Gui&)':
> OP2-Landlord/main.cpp:69:25: warning: catching polymorphic type 'class std::out_of_range' by value [-Wcatch-value=]
>    69 |             catch (std::out_of_range)
>       |                         ^~~~~~~~~~~~
> ```

> ```
> OP2-Landlord/main.cpp:77:25: warning: unused variable 'tset' [-Wunused-variable]
>    77 |             const auto& tset = TileSets.back();
>       |                         ^~~~
> ```

> ```
> OP2-Landlord/main.cpp: In function 'void loadOrCreateTransition(EditorConfig&, Graphics&, Gui&)':
> OP2-Landlord/main.cpp:146:34: warning: catching polymorphic type 'class std::runtime_error' by value [-Wcatch-value=]
>   146 |         catch(std::runtime_error e)
>       |                                  ^
> ```

> ```
> OP2-Landlord/main.cpp: In function 'int main(int, char**)':
> OP2-Landlord/main.cpp:187:27: warning: catching polymorphic type 'class std::exception' by value [-Wcatch-value=]
>   187 |     catch (std::exception e)
>       |                           ^
> ```

> ```
> OP2-Landlord/Graphics.cpp: In member function 'Graphics::Texture Graphics::loadTexturePacked(const void*, size_t) const':
> OP2-Landlord/Graphics.cpp:139:26: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
>   139 |     for (size_t i = 0; i < src->h / 32; ++i)
>       |                        ~~^~~~~~~~~~~~~
> ```

----

Clang shows a subset of the above warnings.

----

Related:
- Issue #96
- PR #114

--- a/OP2-Landlord/Graphics.cpp
+++ b/OP2-Landlord/Graphics.cpp
@@ -136,9 +136,9 @@ Graphics::Texture Graphics::loadTexturePacked(const void* buffer, const size_t b
 
     SDL_Rect sourceRect{ 0, 0, 32, 32 };
     SDL_Rect destRect{ 0, 0, 32, 32 };
-    for (size_t i = 0; i < src->h / 32; ++i)
+    for (int i = 0; i < src->h / 32; ++i)
     {
-        destRect = { (static_cast<int>(i) % 32) * 32, (static_cast<int>(i) / 32) * 32, 32, 32 };
+        destRect = { (i % 32) * 32, (i / 32) * 32, 32, 32 };
         SDL_BlitSurface(src.get(), &sourceRect, destinationSurface.get(), &destRect);
         sourceRect.y += 32;
     }

--- a/OP2-Landlord/Gui.cpp
+++ b/OP2-Landlord/Gui.cpp
@@ -73,8 +73,6 @@ Gui::AppState Gui::initialSetup()
 
     ImGui::SetNextWindowPos(position);
 
-    ImGuiIO& io = ImGui::GetIO();
-
     ImGui::Begin("Initial Setup", nullptr, ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoCollapse);
 
     ImGui::Text("Location of Outpost 2");

--- a/OP2-Landlord/main.cpp
+++ b/OP2-Landlord/main.cpp
@@ -74,8 +74,6 @@ void mainLoop(EditorConfig& config, Graphics& graphics, Gui& gui)
 
         if (!TileSets.empty())
         {
-            const auto& tset = TileSets.back();
-         
             int offset = 250;
             for (const auto& tset : TileSets)
             {

--- a/OP2-Landlord/main.cpp
+++ b/OP2-Landlord/main.cpp
@@ -66,7 +66,7 @@ void mainLoop(EditorConfig& config, Graphics& graphics, Gui& gui)
             {
                 StateTransitionFunctionTable.at(ApplicationState)(config, graphics, gui);
             }
-            catch (std::out_of_range)
+            catch (const std::out_of_range&)
             {
                 std::cout << "[Warning] No transition handler for ApplicationState '" << static_cast<int>(ApplicationState) << "'" << std::endl;
             }
@@ -143,7 +143,7 @@ void loadOrCreateTransition(EditorConfig& config, Graphics& graphics, Gui& gui)
             auto bitmap = Tileset::ReadTileset(*artVol.OpenStream(tsetIndex));
             TileSets.push_back(bmpToTexture(graphics, config, bitmap));          
         }
-        catch(std::runtime_error e)
+        catch(const std::runtime_error& e)
         {
             std::cout << "[Warning] Unable to load tilset '" << artVol.GetName(tsetIndex) << "' : " << e.what() << std::endl;
         }
@@ -184,7 +184,7 @@ int main(int argc, char* argv[])
 
         SDL_Quit();
     }
-    catch (std::exception e)
+    catch (const std::exception& e)
     {
         std::string message(std::string(e.what()) + "\n\nOP2-Landlord will now close.");
 

